stellarphot/core.py

```diff
@@ -62,3 +70,305 @@ def __init__(self, gain=1.0 * u.electron / u.adu,
+    """
+
+    def __init__(self, table_description, data):
+        # Handle parameters
+        self._table_description = table_description
+        self.data = data.copy()
+
+        # Build the data table
+        if not isinstance(self.data, Table):
+            raise TypeError(f"You must provide an astropy Table as data (it is type {type(self.data)}).")
+        else:
+            # Using the BaseTimeSeries class to handle the data table means required columns
+            # are checked.
+            super().__init__(data=self.data)
+
+            # Check that self.data exists now
+            if self.data is None:
+                raise ValueError(f"self.data should exist at this point, but doesn't, crap.")
+
+            # Confirm a proper table description is passed
+            if not isinstance(self._table_description, dict):
+                raise TypeError(f"You must provide a dict as table_description (it is type {type(self._table_description)}).")
+
+            # Check the format of the data table matches the table_description by checking
+            # each column listed in table_description exists and is the correct units.
+            # NOTE: This ignores any columns not in the table_description, it does not remove them.
+            for this_col, this_unit in self._table_description.items():
+                if this_unit is not None:
+                    # Check type
+                    if self.data[this_col].unit != this_unit:
+                        raise ValueError(f"data[{this_col}] is of wrong unit (should be {this_unit} but reported as {self.data[this_col].unit}).")
+
+
+class PhotometryData(BaseEnhancedTable):
+    """
+    A base class to hold reduced photometry data that provides the convience
+    of validating the data table is in the proper format including units.
+    This is based on the `astropy.timeseries.BaseTimeSeries` class to allow
+    inheriting its ability to have required columns (but not requiring the
+    time column).
+
+    Parameters
+    ----------
+
+    observatory: `astropy.coordinates.EarthLocation`
+        The location of the observatory.
+
+    camera: `stellarphot.Camera`
+        A description of the CCD used to perform the photometry.
+
+    data: `astropy.table.Table`
+        A table containing all the instrumental aperture photometry results.
+
+    passband_map: dict, optional (Default: None)
+        A dictionary containing instrumental passband names as keys and
+        AAVSO passband names as values. This is used to automatically
+        update the passband column to AAVSO standard names if desired.
+
+    retain_user_computed: bool, optional (Default: False)
+        If True, any computed columns (see USAGE NOTES below) that already
+        exist in `data` will be retained.  If False, will throw an error
+        if any computed columns already exist in `data`.
+
+    USAGE NOTES: If you input a data file, it MUST contain the following columns
```

Comment:
I still think the "USAGE NOTES" should be in the "Notes" section of the documentation, not here.

Reply:
OK, I moved it.